### PR TITLE
Update docs.  Add charsets and language functionality.  Add parseAll functionality.

### PR DESCRIPTION
**Pull request by mark-bradshaw**
@hueniverse I'd like to take over as maintainer.

### CODE REVIEW

**Comment by mark-bradshaw** [comment]
@hueniverse Hey Eran.  Any word on this?

File: README.md [hunk 1]
```diff
@@ -5,3 +5,62 @@ HTTP Accept-* headers parsing.
...
+
+## Introduction
+
+Accept helps to answer the question of whether the browser can use certain content encoding techniques, such as gzip or deflate.  Accept will parse the headers of a HTTP request and tell you what the preferred encoding is, or if given a list of possible encodings, it will respond with the first available encoding that matches something in the list.
+
+Additional details about Accept headers and content negotiation can be found in [IETF RFC 7231, Section 5.3](https://tools.ietf.org/html/rfc7231#section-5.3).
+
+## Parsing for a specific encoding
+
+Using the `encoding` function will return a `string` matching a type of encoding.  `Encoding` takes two parameters, `encoding(encodingHeader, [preferences])`.  The headers parameter is required, but the preferences are optional.  If no preferences are provided, encoding will find the first supported encoding option and return that.
```
Comment (hueniverse) [inline README.md]: Need to be consistent. Functions should have `encode()` parenthesis and pay attention to case (second encoding is upper case).

**Comment by hueniverse** [comment]
Great start. Need to have both a tutorial format and a reference. This currently lacks the reference (you need to go through the full text for the usage details).

**Comment by mark-bradshaw** [comment]
@hueniverse Thank you for the feedback.  I've make some corrections per your comments, and I'm still interested in maintaining the module.

**Comment by mark-bradshaw** [comment]
@hueniverse I'm wondering if the encoding function shouldn't be changed slightly when it is passed undefined **and** an array of preferences.  According to the spec an undefined (missing) header means any encoding can be used, so if a list of preferences is given, it seems the right thing is to respond with the first item in the preferences array.  What it currently does is respond with an empty string.

This test shows what I think is the correct behavior.  It will fail currently.
```
it('returns first preference when header is undefined', function (done) {

        var accept = Accept.encoding(undefined, ['gzip', 'deflate']);
        expect(accept.isBoom).to.not.exist;
        expect(accept).to.deep.equal('gzip');
        done();
    });
```

The relevant part of the spec is this:

```
A request without an Accept-Encoding header field implies that the
   user agent has no preferences regarding content-codings.  Although
   this allows the server to use any content-coding in a response, it
   does not imply that the user agent will be able to correctly process
   all encodings.
```

**Comment by mark-bradshaw** [comment]
@hueniverse 
I'm adding in some additional functionality now to cover the Accept-Charset header, as well as a convenience function called parseAll that will parse out all Accept-* headers at once.

**Comment by mark-bradshaw** [comment]
BTW, I checked to see how some other accept libraries handle matching an undefined header, and it seems that the current behavior is common (not sure about correct).  So, I'll withdraw my suggestion that encoding() should be modified.

**Comment by mark-bradshaw** [comment]
@hueniverse Just wanted to ping you on this.  Get any comments or see if we can move forward.

**Comment by hueniverse** [comment]
This is way too much for one pull request. How about you close this one and open a few new ones that each add a single new feature. I can't code review this.

As for new functionality, the general rule for hapi modules used by the core framework is to keep them minimalist. I have not looked at the new features yet (still travelling) but make sure they are either simple or enhance hapi.

**Comment by mark-bradshaw** [comment]
Yeah, I went a little overboard while I was waiting for feedback.  I'll split 'em up.

**Comment by hueniverse** [comment]
@mark-bradshaw good energy, just need to package it in smaller pieces :-)

**Comment by mark-bradshaw** [comment]
Sure.  I'm starting a new job this week, so I'm a bit tied up with that,
but I'll swing back around to this.  Might be the following week.

**Pull request closed without merging**